libtenzir/src/execution_node.cpp

```diff
@@ -188,16 +178,21 @@ auto size(const chunk_ptr& chunk) -> uint64_t {
 }
 
 struct metrics_state {
-  auto emit() -> void {
-    values.time_total = std::chrono::duration_cast<duration>(
-      std::chrono::steady_clock::now() - start_time);
+  auto emit(bool exec_node_paused) -> void {
+    if (not exec_node_paused) {
+      values.time_total = std::chrono::duration_cast<duration>(
+        std::chrono::steady_clock::now() - start_time - total_time_paused);
+    }
```

Comment:
Did you consider leaving the total time the wall-clock time, and adding the paused time as a new value in addition to it? What are the pros and cons for how we utilize the metrics?

Reply:
Do you mean a new metric value `time_paused`, while leaving `time_total` as-is?

For the app, this would mean that it would probably have to do the `total - paused` calculation itself to display the proper runtime in the overview page.

Simply calling a `metrics` pipeline for a pipeline that has been paused would then, for those paused durations, display metrics with a `time_total`, but no other noteworthy delta. Someone who would work with the `metrics` output would most likely also have to apply that subtraction.

I guess it depends on what `time_total` actually _means_ - the total _run_ time? If we did that, we would unify the semantics of this particular metric. But I wouldn't mind turning that into the total _life_ time metric, too.

Reply:
From our internal team-wide discussion, it looks like there is a slight consensus towards a separate `time_paused` metric. I'll go ahead and implement that.

Reply:
I implemented a `time_paused_total` metric in https://github.com/tenzir/tenzir/pull/3902/commits/ab740247d7484622b0096064b91c9b7d20203fa0